### PR TITLE
[ALLUXIO-1942] Fix a bug in caching blocks

### PR DESCRIPTION
**Pull request by peisun1115**
Fix a bug in caching blocks

### CODE REVIEW

**Comment by peisun1115** [comment]
@gpang @calvinjia

**Comment by yupeng9** [comment]
Can you explain more on what the bug is and what the fix does?

**Comment by peisun1115** [comment]
@yupeng9 

Without this fix, the client will try to cache the file though the file is in alluxio. This can crash alluxio.

**Comment by yupeng9** [comment]
Can you add a unit test to this?

File: core/client/src/main/java/alluxio/client/file/FileInStream.java [hunk 1]
```diff
@@ -336,43 +336,13 @@ protected boolean shouldCloseCacheStream() {
...
-            mCurrentCacheStream = createCacheStream(currentBlockId, getBlockSize(mPos), address);
-          }
-        } catch (IOException e) {
-          logCacheStreamIOException(e);
-          mShouldCacheCurrentBlock = false;
-        } catch (AlluxioException e) {
-          LOG.warn(BLOCK_ID_NOT_CACHED, currentBlockId, e);
-          mShouldCacheCurrentBlock = false;
-        }
+        updateCacheStream();
```
Comment (gpang) [inline core/client/src/main/java/alluxio/client/file/FileInStream.java]: should this method just take in `currentBlockId` so it doesn't have to compute it again?

**Comment by gpang** [comment]
@peisun1115 thanks for this fix! Would it be possible to verify this somehow with a unitttest?

**Comment by peisun1115** [comment]
@gpang  @yupeng9  Sorry for the delayed response.  I decided to drop this PR since this change is included in  https://github.com/Alluxio/alluxio/pull/3208

About the unit test, I spent sometime looking into it. I am not familiar with Java mock. I decide to add a TODO and open a JIRA ticket for that.  Let me know if that doesn't sound good to you.  thanks

**Pull request closed without merging**